crates/sui-framework/packages/sui-system/sources/sui_system_state_inner.move

```diff
@@ -404,12 +412,7 @@ module sui_system::sui_system_state_inner {
         self: &mut SuiSystemStateInnerV2,
         ctx: &TxContext,
     ) {
-        assert!(
```

Comment:
In theory, yes. In practice, I think not. To be defensive, perhaps what we should do is add a new threshold based on max object size rather than validator count?

Reply:
I looked into this a bit:
- If adding a validator would hit object size limits, the transaction will abort with that error. That's a bit less nice than aborting with `ELimitExceeded`, but the effect will be the same, and it's hard to calculate the exact limit on validator count that the object size limits imply. I think
- On execution limits: `advance_epoch` runs with metering off (I double-checked this by adding an infinite loop inside it and seeing tests that advance epochs diverge). So if the validator set is very big, advance epoch may take longer than expected, but it won't run into gas limits.